nexus/tests/integration_tests/authn_http.rs

```diff
@@ -374,22 +378,37 @@ impl session_cookie::SessionStore for WhoamiServerState {
     type SessionModel = FakeSession;
 
     async fn session_fetch(&self, token: String) -> Option<Self::SessionModel> {
-        self.sessions.lock().unwrap().get(&token).map(|s| *s)
+        self.sessions
+            .lock()
+            .unwrap()
+            .iter()
+            .find(|s| s.token == token)
+            .map(|s| s.clone())
```

Comment:
unimportant style nit: could be:
```suggestion
            .cloned()
```